src/shared/components/results/ResultsView.tsx

```diff
@@ -204,27 +191,15 @@ const ResultsView: FC<ResultsTableProps> = ({
 
   // redirect to entry directly when only 1 result and query marked as "direct"
   useEffect(() => {
-    if (direct && metaData.total === 1 && allResults.length === 1) {
+    if (direct && !hasMoreData && allResults.length === 1) {
       const uniqueItem = allResults[0];
       history.replace(getEntryPathForEntry(uniqueItem));
     }
-  }, [history, direct, metaData, allResults, getEntryPathForEntry]);
-
-  useEffect(() => {
-    if (!data) {
-      return;
-    }
-    const { results } = data;
-    setAllResults((allRes) => [...allRes, ...results]);
-    setMetaData(() => ({
-      total: +(headers?.['x-totalrecords'] || 0),
-      nextUrl: getNextURLFromHeaders(headers),
-    }));
-  }, [data, headers]);
+  }, [history, direct, hasMoreData, allResults, getEntryPathForEntry]);
 
   if (
     // if loading the first page of results
-    (loading && url === initialApiUrl) ||
+    loading ||
```

Comment:
Losing the other part of the test isn't it causing the page to flash on new data being added in? As far as I can see the current version of the hook sets loading to true when any new network call is being made.

Reply:
I had compared it with https://ebi-uniprot.github.io/uniprot-website and didn't notice a difference...although there is a bit of a jump there too, which I guess is because of the "jump to top" PR?

Reply:
What I'm talking about is when scrolling down to get more data, that shouldn't interact with the jump-to-top functionality because there is no client URL change.
If there is no flash when scrolling down to get more data then all good

Reply:
Actually if I slow the network down the results disappear indeed (reason I was talking about the jump to top is that because it does that jump it was not noticeable - but we should fix that too).
Actually because the infinite scroll table handles the display of the loader in that case, I'm thinking to change "loading" to "initialLoading" in the hook and have the url checking logic there. What do you reckon?

Reply:
`initialLoading` sounds good 👍🏽 

Regarding the jump to top issue that you are referring to, I can't seem to see it, maybe I'm not looking at the right place. Do you mind sending me a video of the issue?

Reply:
done